docs/guide/custom-ui.md

```diff
@@ -232,21 +234,24 @@ module.exports = {
 </AposCodeBlock>
 
 ::: warning
-Do not use core actions as your `action` property value - this would lead to unpredictable results and generally broken UI. You may consult what the core actions are in the [AposDocContextMenu component](https://github.com/apostrophecms/apostrophe/blob/main/modules/%40apostrophecms/doc-type/ui/apos/components/AposDocContextMenu.vue) (computed property `menu`).
+Do not use core actions as your `action` property value - this would lead to unpredictable results and generally broken UI. You may consult what the core actions are in the [AposDocContextMenu component logic props](https://github.com/apostrophecms/apostrophe/blob/main/modules/%40apostrophecms/doc-type/ui/apos/logic/AposDocContextMenu.js).
 :::
 
 ::: note
-* The current API supports only `context: "update"` (the custom menu items are available for previously saved documents).
+* The `context`, `action`, `label`, and `modal` properties are required.
+* The current API supports only `context: 'update'` (the custom menu items are available for previously saved documents).
 * The `action` property should be globally unique.
 * Overriding the same `action` is possible (the last wins).
-* You may mark the action as "dangerous" via an optional property `modifiers: [ "danger" ]`.
+* You may mark the action as "dangerous" via an optional property `modifiers: [ 'danger' ]` (see the 'Archive' and 'Unpublish' menu items).
 * An additional optional boolean property `manuallyPublished` is supported. When set to true, the custom menu item is available only for document types that do not have the `autopublish: true` or `localized: false` options set.
 * The `conditions` property is optional. It takes an array of one or more strings specifying conditions that all must be satisfied to determine if the action can be run on the current doc. Valid values are: 'canPublish', 'canEdit', 'canDismissSubmission', 'canDiscardDraft', 'canLocalize', 'canArchive', 'canUnpublish', 'canCopy', 'canRestore'
+* The optional `moduleName` property can be used to invoke the action API on a module other than the defining module.
```

Comment:
There is no "action API" being invoked. If the operation does anything it will be done by the modal. `moduleName` is a prop that is supplied to your modal. If you do not specify `moduleName`, then `moduleName` will be `@apostrophecms/page` for pages. For pieces it will be the appropriate piece type module for that piece.

Reply:
Changed.